src/diffusers/pipelines/stable_diffusion/pipeline_stable_diffusion_inpaint.py

```diff
@@ -32,6 +32,7 @@ def preprocess_mask(mask):
     mask = np.array(mask).astype(np.float32) / 255.0
     mask = np.tile(mask, (4, 1, 1))
     mask = mask[None].transpose(0, 1, 2, 3)  # what does this step do?
+    mask[np.where(mask != 0.0 )] = 1.0 # make sure mask is properly valid
```

Comment:
could we maybe the just convert the mask to bool like this 

```
mask = torch.from_numpy(mask).bool()
maks = (~mask).long()
```

Reply:
In the stable diffusion repo, it looks like this:

```python
mask[mask < 0.5] = 0
mask[mask >= 0.5] = 1
```